Simplenote/Classes/SPNoteListViewController.m

```diff
@@ -340,13 +340,23 @@ - (BOOL)searchControllerShouldBeginSearch:(SPSearchController *)controller
     }
 
     bSearching = YES;
-    [self.tableView reloadData];
+
+    // TODO: Nuke the following as soon as we can!
+    if (!FeatureManager.advancedSearchEnabled) {
+        [self.tableView reloadData];
+    }
     
     return bSearching;
 }
 
 - (void)searchController:(SPSearchController *)controller updateSearchResults:(NSString *)keyword
 {
+    if (FeatureManager.advancedSearchEnabled) {
+        [self.resultsViewController updateSearchResultsWithKeyword:keyword];
+        return;
+    }
```

Comment:
how does this approach compare to the notifications/listeners I've seen elsewhere in the codebase?

would it not be very convenient to add hooks for the listeners here and have one class that initializes behind the feature gate which then hooks into all those listeners?

Reply:
This pattern is [old school delegation pattern](https://developer.apple.com/documentation/swift/cocoa_design_patterns/using_delegates_to_customize_object_behavior). Standard all across Cocoa!

Notifications are usually used to **further** decouple a relationship between two entities. In this particular case, one `SearchController` will **always** have just one delegate, so it's not really necessary to switch over to Notifications.

Notifications are used for scenarios such as: Application became Active / Inactive, and "global" events. **Except** in macOS, where you've got Notifications for scroll events, for instance.

Now, that being said, Notifications are definitely less performant than a delegate call. Wouldn't make any significant mark here, but that's yet another thing to consider!